mayastor/src/bdev/nexus/nexus_channel.rs

```diff
@@ -75,14 +75,21 @@ pub enum DREvent {
 
 impl NexusChannelInner {
     /// very simplistic routine to rotate between children for read operations
-    pub(crate) fn child_select(&mut self) -> usize {
-        debug_assert!(!self.readers.is_empty());
-        if self.previous < self.readers.len() - 1 {
-            self.previous += 1;
+    /// note that the channels can be None during a reconfigure; this is usually
+    /// not the case but a side effect of using the async. As we poll
+    /// threads more often depending on what core we are on etc, we might be
+    /// "awaiting' while the thread is already trying to submit IO.
+    pub(crate) fn child_select(&mut self) -> Option<usize> {
+        if self.readers.is_empty() {
+            None
         } else {
-            self.previous = 0;
+            if self.previous < self.readers.len() - 1 {
+                self.previous += 1;
+            } else {
+                self.previous = 0;
+            }
+            Some(self.previous)
```

Comment:
Just a nit. Looks like this could be replaced by something like this
   self.previous = (self.previous + 1) % self.readers.len();

Reply:
Yes mike, indeed. There is a snag hidden in the grass here, and that's that this checking for 0 should never happen in the first place. Or at least that's my thinking. Regardless your suggestion indeed is much nicer :)

Reply:
I think if self.readers.len() happens to be zero, you get a divide-by-zero error